### PR TITLE
Updating the code for Haystack 2.0-Beta

### PR DESCRIPTION
**Pull request by TuanaCelik**

### CODE REVIEW

File: notion_haystack/notion_exporter.py [hunk 1]
```diff
@@ -38,13 +37,14 @@ def __init__(
             exclude_title_containing=exclude_title_containing,
         )
 
-    def run(self, file_paths: List[str])-> Tuple[Dict, str]:
+    @component.output_types(documents=List[Document])
+    def run(self, page_ids: List[str])-> Tuple[Dict, str]:
```
Comment (bogdankostic) [inline notion_haystack/notion_exporter.py]: The output type annotation doesn't apply anymore for Haystack 2.0
```suggestion
    def run(self, page_ids: List[str]):
```

File: examples/indexing.ipynb [hunk 2]
```diff
@@ -0,0 +1,98 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install nest-asyncio\n",
```
Comment (bogdankostic) [inline examples/indexing.ipynb:20]: Maybe we can add a comment here why `nest-asyncio` is needed?

File: examples/indexing.ipynb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import os\n",
```
Comment (bogdankostic) [inline examples/indexing.ipynb]:
```suggestion
```
`os` is not used in the example.

File: examples/indexing.ipynb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    "\n",
+    "notion_api_key = getpass.getpass(\"Enter Notion API key:\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (bogdankostic) [inline examples/indexing.ipynb:43]: Maybe we can add a comment here that this is a minimal example to just export pages as Haystack Documents.

File: examples/indexing.ipynb [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    "exporter = NotionExporter(api_token=notion_api_key)\n",
+    "exported_pages = exporter.run(page_ids=[\"your_page_id\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (bogdankostic) [inline examples/indexing.ipynb:55]: Maybe we can add a comment here saying that this is a full indexing pipeline that will index the Notion pages directly to the document store.

**Review by bogdankostic — CHANGES_REQUESTED** [review]
Nice, thank you Tuana! Also very cool that you added an example notebook. 💙

This is almost good to go, I just have some minor comments mostly related to adding some explanations to the notebook

File: examples/indexing.ipynb (hunk 2 above)
Comment (TuanaCelik) [reply examples/indexing.ipynb:20]: done

File: examples/indexing.ipynb (hunk 5 above)
Comment (TuanaCelik) [reply examples/indexing.ipynb:55]: done

**Pull request merged**